"Create folder" button for save dialog

wolfreak99 opened the pull request:
"platform_ensure_directory_exists" forces a directory to be created and i realized that if someone wants to create a folder, they need to use the system dialog window, which will disconnect any network games they are in.

janisozaur commented on data/language/en-GB.txt:
```diff
@@ -4232,6 +4232,9 @@ STR_5920    :Render weather effects
 STR_5921    :{SMALLFONT}{BLACK}If enabled, rain and gloomy colours will be rendered during storms.
 STR_5922    :{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{SMALLFONT}{BLACK}Max {STRINGID}
 STR_5923    :{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{SMALLFONT}{BLACK}Max {COMMA16} {STRINGID} per train
+STR_5924    :New folder
```
this should be named "directory"

AaronVanGeffen replied:
Both Windows and macOS use 'folder' to describe directories to the user. Actually, even KDE and GNOME do, too. So, it makes sense to use use 'folder', really.

janisozaur replied:
`$CWD`, `cd`, `dir`, and just a handful of ["folder"](https://github.com/torvalds/linux/search?utf8=%E2%9C%93&q=folder) vs ["directory"](https://github.com/torvalds/linux/search?utf8=%E2%9C%93&q=directory) in public *filesystem* sources all point to "directory".

AaronVanGeffen replied:
Command line tools versus graphical user interface. I'm not disputing the use of 'directory' internally -- it is the technical term for it, no doubt.

The file / folder analogy is used consistently in GUIs though -- not just in strings, but also in icons. I do believe we should follow that -- the consistency argument works that way, too.

IntelOrca replied:
@janisozaur Typically they are refereed to as directories on a command line and folders in graphical user interfaces.

https://en.wikipedia.org/wiki/Directory_(computing)#Folder_metaphor

janisozaur commented on src/windows/loadsave.c:
```diff
@@ -38,27 +38,29 @@ enum {
+	WIDX_DEFAULT,
 	WIDX_UP,
-	WIDX_NEW,
+	WIDX_NEW_FOLDER,
```
`WIDX_NEW_DIRECTORY`

Gymnasiast commented on data/language/en-GB.txt:
```diff
@@ -4232,6 +4232,9 @@ STR_5920    :Render weather effects
 STR_5922    :{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{SMALLFONT}{BLACK}Max {STRINGID}
 STR_5923    :{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{SMALLFONT}{BLACK}Max {COMMA16} {STRINGID} per train
+STR_5924    :New folder
+STR_5925    :Type the name of the folder.
```
I would change this to `Type the name of the folder.`

Gymnasiast commented on src/windows/loadsave.c:
```diff
@@ -389,43 +394,71 @@ static void window_loadsave_scrollmouseover(rct_window *w, int scrollIndex, int
 
 static void window_loadsave_textinput(rct_window *w, int widgetIndex, char *text)
 {
+	if (text == NULL || text[0] == 0)
```
Could you move this under the declaration again? It's convention to have these either locally or at the top of the function.

Gymnasiast commented on src/windows/loadsave.c:
```diff
@@ -389,43 +394,71 @@ static void window_loadsave_scrollmouseover(rct_window *w, int scrollIndex, int
-	safe_strcpy(path, _directory, sizeof(path));
-	safe_strcat_path(path, text, sizeof(path));
-	path_append_extension(path, _extension, sizeof(path));
+		safe_strcpy(path, _directory, sizeof(path));
```
These three lines need extra indentation.

Gymnasiast commented:
I left a few remarks, but it otherwise looks good codewise. When you have changed the bits, could you rebase it? After that I'll test it.

wolfreak99 commented:
@Gymnasiast  I fixed the remarks and rebased it, i havent had a chance yet to test it but i'm about to go into work, if you can test compilation before i get home (in about 8 hours from the time this is posted) cool, if not, i'll test it and let you know you can test it after i get home.

Gymnasiast commented on src/windows/loadsave.c:
```diff
@@ -547,12 +580,13 @@ static void window_loadsave_populate_list(rct_window *w, int includeNewItem, con
 	_listItemsCount = 0;
 
 	// Show "new" button when saving
```
```Show "new" buttons when saving```

Gymnasiast commented:
Hm, it works, but I feel the toolbar is getting a bit cluttered. I think the window would benefit from using icon rather than text. The problem is that I don't think we have suitable sprites at the moment.
Any ideas, @IntelOrca @duncanspumpkin @Broxzier ?

IntelOrca commented:
@Gymnasiast I don't have any ideas

Gymnasiast commented:
Hm, ok. Well, I'm not going to hold back this PR for it, but I will create an issue for it.
@wolfreak99 If you could address the two comments I left a line note about, this PR will be merged.

Broxzier commented:
@Gymnasiast: I agree that the save game dialog could use some work, but this PR is in line with what is already there. Tidying it up is another task.

wolfreak99 commented:
@Gymnasiast  I have fixed the comments. I have also merged it all down to one commit as to not clog the history with "Formatting improvements" etc.
